mne/io/edf/edf.py

```diff
@@ -342,12 +342,13 @@ def _read_header(fname, exclude):
     ext = os.path.splitext(fname)[1][1:].lower()
     logger.info('%s file detected' % ext.upper())
     if ext in ('bdf', 'edf'):
-        return _read_edf_header(fname, exclude)
-    elif ext in ('gdf'):
-        return _read_gdf_header(fname, exclude), None
+        edf_info, orig_units = _read_edf_header(fname, exclude)
+    elif ext == 'gdf':
+        edf_info, orig_units = _read_gdf_header(fname, exclude), None
     else:
-        raise NotImplementedError(
-            'Only GDF, EDF, and BDF files are supported, got %s.' % ext)
+        raise NotImplementedError(f'Only GDF, EDF, and BDF files are supported'
+                                  f', got {ext}.')
+    return edf_info, orig_units
```

Comment:
If you're moving the check to `_get_info` (which I think is a great idea), then you can also revert my changes here and directly `return` without creating two names.